File: src/main/java/com/workforce/management/service/ShiftService.java
package com.workforce.management.service;

import com.workforce.management.dto.ShiftDto;

import java.util.List;

public interface ShiftService {

    ShiftDto createShift(ShiftDto shiftDto);

    ShiftDto updateShift(Long shiftId, ShiftDto shiftDto);

    void deleteShift(Long shiftId);

    List<ShiftDto> getAllShifts(Long employeeId); // employeeId optional filter

    List<ShiftDto> getMyShifts();
}

File: workforce-management-frontend/src/services/ShiftService.js
import api from "./api";

const SHIFT_API_BASE_URL = "/api/shifts";

export const getMyShifts = () => api.get(`${SHIFT_API_BASE_URL}/my`);

export const getAllShifts = (employeeId) => {
  if (employeeId) {
    return api.get(SHIFT_API_BASE_URL, { params: { employeeId } });
  }
  return api.get(SHIFT_API_BASE_URL);
};

export const createShift = (shift) => api.post(SHIFT_API_BASE_URL, shift);

export const updateShift = (shiftId, shift) => api.put(`${SHIFT_API_BASE_URL}/${shiftId}`, shift);

export const deleteShift = (shiftId) => api.delete(`${SHIFT_API_BASE_URL}/${shiftId}`);